Mathlib/Algebra/Hom/Centroid.lean

```diff
@@ -514,4 +515,43 @@ def commRing (h : ∀ a b : α, (∀ r : α, a * r * b = 0) → a = 0 ∨ b = 0)
 
 end NonUnitalRing
 
+
+section StarRing
+
+variable [NonUnitalNonAssocSemiring α] [StarRing α]
+
+instance : Star (CentroidHom α) :=
+{
+  star := fun f => {
+    toFun := fun a => star (f (star a))
+    map_zero' := by
+      simp only [star_zero, map_zero]
+    map_add' := fun a b => by simp only [star_add, map_add]
+    map_mul_left' := fun a b => by simp only [star_mul, map_mul_right, star_star]
+    map_mul_right' := fun a b => by simp only [star_mul, map_mul_left, star_star]
+  }
+
+}
+
+
+@[simp] lemma star_apply (f : CentroidHom α) (a : α) : (star f) a = star (f (star a)) := rfl
+
+/--
+Let α be a star ring with commutative centroid. Then the centroid is a star ring.
+-/
+def star_ring (mul_comm : ∀ f g : CentroidHom α, f * g = g * f) : StarRing (CentroidHom α) :=
+{ star_involutive := fun f => by
+    ext
+    rw [star_apply, star_apply, star_star, star_star]
```

Comment:
This and `star_add` doesn't need commutativity; can you promote the instance above to `StarAddMonoid`?

Reply:
Is your proposal that we change the definition of `CentroidHom` from
```
tructure CentroidHom (α : Type*) [AddMonoid α] [Mul α] extends α →+ α where...
```
to
```
structure CentroidHom (α : Type*) [AddMonoid α] [Mul α] extends α →+ α where...
```
?

Reply:
Not at all, it's that you add a separate `StarAddMonoid` instance

Reply:
Ah, I see what you mean. Done.